### PR TITLE
feat!: use shadow root (ACT-004)

### PR DESCRIPTION
**Pull request by DecathectZero**
Shadow DOM allows the segregation of styling without the full isolation of an iframe. Shadow DOM is a widely supported technology now, as long as you’re not on internet explorer. 
https://caniuse.com/shadowdomv1

I know production webchats are expected to be used across a wide variety of browsers, we could just detect if they are on an outdated browser and use regular DOM and compromise the styling. This is likely not a simple undertaking, and is an extensive refactor of react-chat. But I would rather do this now and not later as we build more and more functionality into the chat.

## extensibility improvements
It will be much easier for others to take the chat and modify it, now that it is all within a single bundle.

### technical updates
I had to fork `stitches` as it was not longer being maintained. It's crucial that CSS variables are attached to `:host` and not `:root` for shadowDOM to work: https://github.com/voiceflow/stitches/commit/bb6d5818c563a24ce4b2d67a0ea3c7cfef120918

The ci will temporarily point to `https://cdn.voiceflow.com/widget/alpha`. We will continue using the `/widget/` path.

BREAKING CHANGE: the widget repo is fully removed, and react-chat is no longer added within an iframe.

### CODE REVIEW

File: packages/react-chat/src/styles/theme.ts [hunk 1]
```diff
@@ -21,6 +24,8 @@ export interface FontOptions {
 }
 
 export const { styled, config, keyframes } = createStitches({
+  ...{ root: shadowRoot },
```
Comment (DecathectZero) [inline packages/react-chat/src/styles/theme.ts:27]: spread for some typing issue with stitches, it does accept this.

File: .circleci/continue-config.yml [hunk 2]
```diff
@@ -58,12 +58,7 @@ jobs:
       - vfcommon/clone_s3_assets:
           step_name: Uploading React Chat to CDN
           from: packages/react-chat/dist
-          to: s3://cdn.voiceflow.com/react-chat
-          clean_destination: false
-      - vfcommon/clone_s3_assets:
-          step_name: Uploading Widget to CDN
-          from: packages/widget/dist
-          to: s3://cdn.voiceflow.com/widget
+          to: s3://cdn.voiceflow.com/widget/alpha # TODO: change to production after validation
```
Comment (effervescentia) [inline .circleci/continue-config.yml:61]: should be changed back now?

File: packages/react-chat/src/contexts/RuntimeContext/index.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React, { createContext, useMemo } from 'react';
+
+import { Settings, useRuntimeState } from './useRuntimeState';
+
+// split up API and state to prevent unnecessary re-renders
```
Comment (effervescentia) [inline packages/react-chat/src/contexts/RuntimeContext/index.tsx:5]: ❤️

File: packages/react-chat/src/contexts/RuntimeContext/useRuntimeState.ts [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+      close,
+      addTurn,
+      feedback: runtime.saveFeedback,
+      setStatus,
+      isStatus,
+      reset,
+
+      // these are meant to be static, so bundling them with the API
+      assistant,
+      config,
```
Comment (effervescentia) [inline packages/react-chat/src/contexts/RuntimeContext/useRuntimeState.ts:171]: we could add a separate `meta` section or something for static stuff that isn't state

**Review by effervescentia — APPROVED** [review]

File: .circleci/continue-config.yml (hunk 2 above)
Comment (DecathectZero) [reply .circleci/continue-config.yml:61]: I want to still push it here to test the CI/CD pipeline. After the first push on master works, I'll switch it back.

**Pull request merged**